acceptance/invoke/pack.go

```diff
@@ -0,0 +1,273 @@
+// +build acceptance
+
+package invoke
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/Masterminds/semver"
+
+	acceptanceOS "github.com/buildpacks/pack/acceptance/os"
+	h "github.com/buildpacks/pack/testhelpers"
+)
+
+type PackInvoker struct {
+	testObject      *testing.T
+	path            string
+	home            string
+	dockerConfigDir string
+	fixtureManager  PackFixtureManager
+}
+
+type packPathsProvider interface {
+	Path() string
+	FixturePaths() []string
+}
+
+func NewPackInvoker(testObject *testing.T, packAssets packPathsProvider, dockerConfigDir string) *PackInvoker {
+	testObject.Helper()
+
+	home, err := ioutil.TempDir("", "buildpack.pack.home.")
+	if err != nil {
+		testObject.Fatalf("couldn't create home folder for pack: %s", err)
+	}
+
+	return &PackInvoker{
+		testObject:      testObject,
+		path:            packAssets.Path(),
+		home:            home,
+		dockerConfigDir: dockerConfigDir,
+		fixtureManager: PackFixtureManager{
+			testObject: testObject,
+			locations:  packAssets.FixturePaths(),
+		},
+	}
+}
+
+func (i *PackInvoker) Cleanup() {
+	i.testObject.Helper()
+
+	err := os.RemoveAll(i.home)
+	h.AssertNil(i.testObject, err)
+}
+
+func (i *PackInvoker) cmd(name string, args ...string) *exec.Cmd {
+	i.testObject.Helper()
+
+	cmdArgs := append([]string{name}, args...)
+	cmdArgs = append(cmdArgs, "--no-color")
+	if i.Supports("--verbose") {
+		cmdArgs = append(cmdArgs, "--verbose")
+	}
+
+	cmd := i.baseCmd(cmdArgs...)
+
+	cmd.Env = append(os.Environ(), "DOCKER_CONFIG="+i.dockerConfigDir)
+	if i.home != "" {
+		cmd.Env = append(cmd.Env, "PACK_HOME="+i.home)
+	}
+
+	return cmd
+}
+
+func (i *PackInvoker) baseCmd(parts ...string) *exec.Cmd {
+	return exec.Command(i.path, parts...)
+}
+
+func (i *PackInvoker) Run(name string, args ...string) (string, error) {
+	i.testObject.Helper()
+
+	output, err := i.cmd(name, args...).CombinedOutput()
+
+	return string(output), err
+}
+
+func (i *PackInvoker) RunSuccessfully(name string, args ...string) string {
+	i.testObject.Helper()
+
+	output, err := i.Run(name, args...)
+	h.AssertNil(i.testObject, err)
+
+	return output
+}
+
+func (i *PackInvoker) JustRunSuccessfully(name string, args ...string) {
```

Comment:
I appreciate this change, rather than the `PackWithOutput` pattern before. Thanks!